test/extractor/ConfigurationExtractor.test.ts

```diff
@@ -0,0 +1,128 @@
+import { cloneDeep } from 'lodash';
+import { fshtypes } from 'fsh-sushi';
+import { ConfigurationExtractor } from '../../src/extractor';
+import { Package } from '../../src/processor';
+import {
+  ExportableProfile,
+  ExportableCaretValueRule,
+  ExportableExtension,
+  ExportableValueSet,
+  ExportableCodeSystem
+} from '../../src/exportable';
+
+describe('ConfigurationExtractor', () => {
+  let resources: Package;
+
+  beforeAll(() => {
+    resources = new Package();
+    const typicalProfile = new ExportableProfile('TypicalProfile');
+    const profileUrl = new ExportableCaretValueRule('');
+    profileUrl.caretPath = 'url';
+    profileUrl.value = 'http://example.org/our/ig/StructureDefinition/TypicalProfile';
+    typicalProfile.rules.push(profileUrl);
+
+    const typicalExtension = new ExportableExtension('TypicalExtension');
+    const extensionUrl = cloneDeep(profileUrl);
+    extensionUrl.value = 'http://example.org/our/ig/StructureDefinition/TypicalExtension';
+    const extensionStatus = new ExportableCaretValueRule('');
+    extensionStatus.caretPath = 'status';
+    extensionStatus.value = new fshtypes.FshCode('draft');
+    const extensionVersion = new ExportableCaretValueRule('');
+    extensionVersion.caretPath = 'version';
+    extensionVersion.value = '0.8.6';
+    const extensionFhirVersion = new ExportableCaretValueRule('');
+    extensionFhirVersion.caretPath = 'fhirVersion';
+    extensionFhirVersion.value = '4.0.1';
+    typicalExtension.rules.push(extensionUrl, extensionStatus, extensionVersion);
+
+    const typicalValueSet = new ExportableValueSet('TypicalVS');
+    const valueSetUrl = cloneDeep(profileUrl);
+    valueSetUrl.value = 'http://example.org/our/ig/ValueSet/TypicalVS';
+    typicalValueSet.rules.push(
+      valueSetUrl,
+      cloneDeep(extensionStatus),
+      cloneDeep(extensionVersion),
+      cloneDeep(extensionFhirVersion)
+    );
+
+    const typicalCodeSystem = new ExportableCodeSystem('TypicalCodes');
+    const codeSystemUrl = cloneDeep(profileUrl);
+    codeSystemUrl.value = 'http://example.org/our/ig/CodeSystem/TypicalCodes';
+    const codeSystemStatus = cloneDeep(extensionStatus);
+    codeSystemStatus.value = new fshtypes.FshCode('active');
+    typicalCodeSystem.rules.push(
+      codeSystemUrl,
+      codeSystemStatus,
+      cloneDeep(extensionVersion),
+      cloneDeep(extensionFhirVersion)
+    );
+
+    resources.add(typicalProfile);
+    resources.add(typicalExtension);
+    resources.add(typicalValueSet);
+    resources.add(typicalCodeSystem);
+  });
+
+  describe('#process', () => {
+    it('should return a configuration with a canonical and fhirVersion when there are no resources', () => {
```

Comment:
Wait.  But how?!

Reply:
OK.  I debugged it.  The `canonical` and `fhirVersion` are both set to `''`.  This doesn't really seem reasonable to me...  I mean, it's an edge case for sure, but what's the point?  I'd rather that we either don't return a `Configuration` at all or we just hard-code some values in this case (some example `canonical` and `4.0.1` for the `fhirVersion`).